homeassistant/components/ecovacs_mqtt/config_flow.py

```diff
@@ -0,0 +1,271 @@
+                    }
+                ),
+                user_input or {CONF_COUNTRY: self.hass.config.country},
+            ),
+            errors=errors,
+        )
+
+    async def async_step_options(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the options step."""
+
+        errors = {}
+        if user_input is not None:
+            if len(user_input[CONF_DEVICES]) < 1:
+                errors[CONF_DEVICES] = "select_robots"
+            else:
+                return self.async_create_entry(
+                    title=self._data[CONF_USERNAME],
+                    data=self._data,
+                    options=user_input,
+                )
+
+        return self.async_show_form(
+            step_id="options",
+            data_schema=_get_options_schema(self._devices, {}),
+            errors=errors,
+        )
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle initiation of re-authentication."""
+        self._entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        self._data = dict(entry_data)
+        return await self.async_step_auth(entry_data)
+
+
+def _get_options_schema(
+    devices: list[DeviceInfo], defaults: Mapping[str, Any]
+) -> vol.Schema:
+    """Return options schema."""
+    select_options = []
+
+    for entry in devices:
+        api_info = entry.api_device_info
+        name = api_info["name"]
+        label = api_info.get("nick", name)
+        select_options.append(selector.SelectOptionDict(value=name, label=label))
+
+    return vol.Schema(
+        {
+            vol.Required(
+                CONF_DEVICES, default=defaults.get(CONF_DEVICES, vol.UNDEFINED)
+            ): selector.SelectSelector(
+                selector.SelectSelectorConfig(
+                    options=select_options,
+                    multiple=True,
+                )
+            )
+        }
+    )
+
+
+async def _retrieve_devices(
+    hass: HomeAssistant, integration_config: Mapping[str, Any]
+) -> list[DeviceInfo]:
+    verify_ssl = integration_config.get(CONF_VERIFY_SSL, True)
+    deebot_config = Configuration(
+        aiohttp_client.async_get_clientsession(hass, verify_ssl=verify_ssl),
+        device_id=get_client_device_id(hass, integration_config[CONF_MODE]),
+        country=integration_config[CONF_COUNTRY],
+        verify_ssl=verify_ssl,
+    )
+
+    authenticator = Authenticator(
+        deebot_config,
+        integration_config[CONF_USERNAME],
+        md5(integration_config[CONF_PASSWORD]),
+    )
+    api_client = ApiClient(authenticator)
+
+    return await api_client.get_devices()
+
+
+class EcovacsOptionsFlowHandler(OptionsFlow):
+    """Handle ecovacs options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self._config_entry = config_entry
```

Comment:
And if you are removing the device list, might as well look at OptionsFlowWithConfigEntry